src/Transports/MassTransit.WebJobs.ServiceBusIntegration/MessageReceiver.cs

```diff
@@ -18,8 +18,9 @@ public class MessageReceiver :
     {
         readonly IAsyncBusHandle _busHandle;
         readonly IServiceBusHostConfiguration _hostConfiguration;
-        readonly ConcurrentDictionary<string, IBrokeredMessageReceiver> _receivers;
+        readonly ConcurrentDictionary<string, Lazy<IBrokeredMessageReceiver>> _receivers;
         readonly IBusRegistrationContext _registration;
+        readonly object _configureLock = new object();
```

Comment:
I don't believe that this lock is necessary with the `Lazy` constructor specified in `GetOrAdd` factory.

Reply:
Thanks for reviewing @henkwedel. The `_configureLock` protects the scenario of concurrent calls where the key in the `ConcurrentHashSet` differs between the calls (e.g. message received from different queues at the same time).

It's *almost* enough to have just the `_configureLock` and not the `Lazy`, except that the lambda passed to `GetOrAdd` might get invoked twice for the same configuration. Then the underlying bus config methods are called twice. Even though the calls aren't concurrent it seems that duplicate calls to the config methods is not something that's expected (e.g. some add an element to a list without checking if it's already present).

Reply:
Ah, OK - I must not be familiar enough with the code base.

> [..] except that the lambda passed to GetOrAdd might get invoked twice for the same configuration.

Hmm, that's not my understanding of what would happen though. I believe `GetOrAdd` will discard one of the `Lazy` instances in the duplicate call scenarios and since `.Value` isn't invoked on the discarded instance the lambda is not evaluated twice.

Reply:
Yep that's right, the `Lazy` prevents the double evaluation. If there was only `_configureLock` then double evaluation could happen.